src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs

```diff
@@ -63,6 +68,8 @@ public Task AddAsync(FunctionInstanceLogEntry item, CancellationToken cancellati
 
         private void StartFunction(FunctionInstanceLogEntry item)
         {
+            _hostMetrics.IncrementActiveFunctionCount();
```

Comment:
There's an existing IFunctionActivityStatusProvider service implemented in the WebJobs SDK that tracks this information already. You should use that. It also accounts for retries - i.e. even if no invocations are active, we might be processing retries for an invocation.

Reply:
See here for how we use this https://github.com/Azure/azure-functions-host/blob/92d51f0fdf842af8dec3f1ea3275785ec89d3819/src/WebJobs.Script.WebHost/Controllers/HostController.cs#L220

Reply:
@mathewc I tried this approach out and within the EndFunction() method, the function status returns `OutstandingInvocations=1` and at this stage I would expect it to be 0 when the function completes. Is there a better place for me to be incrementing and decrementing the `ActiveInvocationCount` than here?

Reply:
I think the difficulty is that you're attempting to maintain a separate incremented count for this. Instead, you'd just call IFuncitonActivityStatusProvider at the point in time when you need to get the current count and log it. Not each time a function ends. The platform is already maintaining the count [here](https://github.com/Azure/azure-webjobs-sdk/blob/1f5c638d08a15360e54fd620d34276b10f186da6/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs#L160).

Reply:
Fixed by moving this to the host metric provider and not using .net metrics to capture this value.